libqtile/backend/x11/xcbq.py

```diff
@@ -759,9 +760,57 @@ def query_tree(self):
             parent = Window(self.conn, q.parent)
         return root, parent, [Window(self.conn, i) for i in q.children]
 
-    def paint_borders(self, color):
-        if color:
-            self.set_attribute(borderpixel=self.conn.color_pixel(color))
+    def paint_borders(self, colors, borderwidth, width, height):
```

Comment:
I agree, and I have a branch where I am merging these two classes (more specifically, forming a template Window for a base class in the backend base, subclassed in the x11 backend to actually implement the X specific parts). Cleaning up things like this will definitely play a part :)